### DISCUSSION TITLE
Setting up peer identifiers with 4 nodes on the cloud

### DISCUSSION THREAD

**Discussion by steveodhiambo** [tendermint/tendermint#8492] [Q&A]
Hello, I followed the following tutorial [https://docs.tendermint.com/master/introduction/quick-start.html](tendermint) on how to setup a cluster of nodes. I currently have set up 4 ubuntu cloud machines and installed tendermint. I ran the following commands 
`tendermint show_node_id --home ./mytestnet/node0
tendermint show_node_id --home ./mytestnet/node1
tendermint show_node_id --home ./mytestnet/node2
tendermint show_node_id --home ./mytestnet/node3
`
followed by the following command with their respective ID's and IP's
`tendermint start --home ./mytestnet/node0 --proxy-app=kvstore --p2p.persistent-peers="ID1@IP1:26656,ID2@IP2:26656,ID3@IP3:26656,ID4@IP4:26656"
tendermint start --home ./mytestnet/node1 --proxy-app=kvstore --p2p.persistent-peers="ID1@IP1:26656,ID2@IP2:26656,ID3@IP3:26656,ID4@IP4:26656"
tendermint start --home ./mytestnet/node2 --proxy-app=kvstore --p2p.persistent-peers="ID1@IP1:26656,ID2@IP2:26656,ID3@IP3:26656,ID4@IP4:26656"
tendermint start --home ./mytestnet/node3 --proxy-app=kvstore --p2p.persistent-peers="ID1@IP1:26656,ID2@IP2:26656,ID3@IP3:26656,ID4@IP4:26656"
`
however I get the following error message

`ERROR: failed to create node: error="failed to create peer manger:failed to add peer .... cant add self to peer store"`
how can I fix this issue

**Comment by creachadair** [comment]
It looks as if you are passing the same four addresses as the persistent peers for all four nodes. What the error means is that this leads to each node trying to add _itself_ as a persistent peer, which is not allowed.

It should suffice to remove `ID1@IP1:26656` from the first node's command line, `ID2@IP2:26656` from the second, etc., when starting them up.

Or you can add those settings directly to each `config.toml` so that you needn't type them out each time to start up the nodes.

**Reply by steveodhiambo** [reply]
thank you i will try it and see how it works
